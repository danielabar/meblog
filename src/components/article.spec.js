import React from "react"
import { render } from "@testing-library/react"
import "@testing-library/jest-dom"

import Article from "./article"

describe("Article", () => {
  it("renders correctly", () => {
    const container = render(
      <Article
        key="123abc"
        id="123abc"
        to="/blog/some-article"
        title="Some Article"
        category="Rails"
        date="August 2021"
        excerpt="First sentence of the article blah blah blah..."
      />
    )
    expect(container).toMatchSnapshot()
  })
})
