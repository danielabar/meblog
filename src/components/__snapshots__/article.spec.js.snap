// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <article
        class="container"
      >
        <div
          class="subheader"
        >
          <div
            class="date"
          >
            August 2021
          </div>
          <div
            class="category"
          >
            Rails
          </div>
        </div>
        <a
          class="title"
          href="/blog/some-article"
        >
          Some Article
        </a>
        <div
          class="excerpt"
        >
          First sentence of the article blah blah blah...
        </div>
        <a
          class="read"
          href="/blog/some-article"
        >
          Read
        </a>
      </article>
    </div>
  </body>,
  "container": <div>
    <article
      class="container"
    >
      <div
        class="subheader"
      >
        <div
          class="date"
        >
          August 2021
        </div>
        <div
          class="category"
        >
          Rails
        </div>
      </div>
      <a
        class="title"
        href="/blog/some-article"
      >
        Some Article
      </a>
      <div
        class="excerpt"
      >
        First sentence of the article blah blah blah...
      </div>
      <a
        class="read"
        href="/blog/some-article"
      >
        Read
      </a>
    </article>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
